File: Project/DBProject/ClientApp/src/app/shared/models/weather.model.ts
import { Main } from "./main";
import { Weather } from "./Weather";
import { Wind } from "./Wind";
import { Clouds } from "./Clouds";
import { Sys } from "./Sys";
import Coord from "./Coord";

export class WeatherModel {
  coord: Coord;
  main: Main;
  weather: Weather[];
  wind: Wind;
  clouds: Clouds;
  sys: Sys;
  visibility: number;
  dt: number;
  id: number;
  name: string;
}

File: Project/DBProject/Models/WeatherModel.cs
using DBProject.Models.WeatherDetails;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace DBProject.Models
{
    public class WeatherModel
    {
        public WeatherModel() { }
        [NotMapped]
        public Coord Coord { get; set; }
        [NotMapped]
        public Weather[] Weather { get; set; }
        [NotMapped]
        public Main Main { get; set; }
        [NotMapped]
        public Wind Wind { get; set; }
        [NotMapped]
        public Clouds Clouds { get; set; }
        [NotMapped]
        public Sys Sys { get; set; }
        [NotMapped]
        public int Visibility { get; set; }

        public double Dt { get; set; }
        public int Id { get; set; }
        public string Name { get; set; }
        public string Timezone { get; set; }
    }
}
